src/odemis/gui/cont/temperature.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+
+"""
+@author: Anders Muskens
+
+Copyright © 2020 Anders Muskens, Delmic
+
+Handles the switch of the content of the main GUI tabs.
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License version 2 as published by the Free
+Software Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+
+"""
+
+import wx
+import collections
+import numpy
+import time
+import logging
+from odemis.util import RepeatingTimer
+import odemis.model as md
+
+
+class TemperatureController(object):
+    '''
+    Temperature Controller
+    Displays the temperature from a thermostat component in the main GUI on a predefined control
+    and updates at a set sampling period, which is polled.
+    '''
+
+    def __init__(self, main_frame, thermostat, sampling_period=1.0, len_history=5):
+        '''
+        main_frame: (wx) main xrc frame of the GUI
+        thermostat: (HwComponent) a component with the VA "temperature"
+        sampling_period: (float) the sampling period of the controller, in seconds.
+            Will update the temperature at this period
+        len_history: (positive int) the length of the temperature history queue.
+            This queue contains the last sampled measurements of temperature and is
+            used to calculate the rate of change as a linear regression
+        '''
+
+        if not hasattr(thermostat, "temperature"):
+            raise ValueError("Thermostat is missing the temperature VA")
+        self.thermostat = thermostat
+
+        self.temperature_label = main_frame.temperature_display
+        self.temperature_label.Show()
+
+        self._temperature_sample_period = sampling_period  # in seconds
+        self._len_temp_q = len_history
+
+        self._time_last_warning = 0  # time of the last warning
+        self._time_last_speed_warning = 0
+
+        val = self.thermostat.temperature.value
+        self.temperature_history = collections.deque(self._len_temp_q * [val], self._len_temp_q)
+        self._update_temperature_timer = RepeatingTimer(self._temperature_sample_period, self._updateTemperature)
+        self._update_temperature_timer.start()
+
+    def _updateTemperature(self):
+        new_temperature = self.thermostat.temperature.value
```

Comment:
Let's call this just `temperature`, and everyone should assume this is the current temperature.